tests/common/devices/sonic.py

```diff
@@ -1189,7 +1190,25 @@ def get_extended_minigraph_facts(self, tbinfo, namespace = DEFAULT_NAMESPACE):
         except (ValueError, KeyError):
             pass
 
+        # set 'backend' flag for mg_facts
+        # a 'backend' topology may has different name convention for some parameter
+        self.update_backend_flag(tbinfo, mg_facts)
+
         return mg_facts
+    
+    def update_backend_flag(self, tbinfo, mg_facts):
+        mg_facts[constants.IS_BACKEND_TOPOLOGY_KEY] = self.assert_topo_is_backend(tbinfo)
+
+    # assert whether a topo is 'backend' type
+    def assert_topo_is_backend(self, tbinfo):
+        topo_key = constants.TOPO_KEY
+        name_key = constants.NAME_KEY
+        if topo_key in tbinfo.keys() and name_key in tbinfo[topo_key].keys():
+            topo_name = tbinfo[topo_key][name_key]
+            if constants.BACKEND_TOPOLOGY_FLAG in topo_name:
+                return True
+        else:
```

Comment:
you can get rid of this 'else' and just return False

Reply:
Done